wgsl/index.bs

```diff
@@ -1728,33 +1750,38 @@ together with an interpretation of the contents of those locations as a [SHORTNA
     <tr><th>Constraint<th>Type<th>Description
   </thead>
   <tr algorithm="memory reference type">
-    <td>|SC| is a [=storage class=],<br>|T| is a [=storable=] type
-    <td>ref&lt;|SC|,|T|&gt;
+    <td style="width:25%">|SC| is a [=storage class=],<br>|T| is a [=storable=] type,<br>|A| is an [=access mode=]
+    <td>ref&lt;|SC|,|T|,|A|&gt;
     <td>The <dfn noexport>reference type</dfn>
-        identified with the set of [=memory views=] for memory locations in |SC| holding values of type |T|.<br>
+        identified with the set of [=memory views=] for memory locations in |SC| holding values of type |T|,
+        supporting memory accesses described by mode |A|.<br>
         In this context |T| is known as the <dfn noexport>store type</dfn>.<br>
-        Reference types are not written [SHORTNAME] progam source; instead they are used to analyze a [SHORTNAME] program.
+        Reference types are not written in [SHORTNAME] progam source;
+        instead they are used to analyze a [SHORTNAME] program.
   <tr algorithm="pointer type">
-    <td>|SC| is a [=storage class=],<br>|T| is a [=storable=] type
-    <td>ptr&lt;|SC|,|T|&gt;
+    <td>|SC| is a [=storage class=],<br>|T| is a [=storable=] type,<br>|A| is an [=access mode=]
+    <td>ptr&lt;|SC|,|T|,|A|&gt;
     <td>The <dfn noexport>pointer type</dfn>
-        identified with the set of [=memory views=] for memory locations in |SC| holding values of type |T|.<br>
+        identified with the set of [=memory views=] for memory locations in |SC| holding values of type |T|,
+        supporting memory accesses described by mode |A|.<br>
         In this context |T| is known as the <dfn noexport>pointee type</dfn>.<br>
-        Pointer types appear in [SHORTNAME] progam source.
+        Pointer types may appear in [SHORTNAME] progam source.
 </table>
 
-<div class='example wgsl' heading='Pointer type (program fragment)'>
+<div class='example wgsl' heading='Pointer type'>
   <xmp highlight='rust'>
     fn my_function(
-      // 'ptr<function,i32>' is the type of a pointer value that references storage
-      // for keeping an 'i32' value, using memory locations in the 'function' storage
-      // class.  Here 'i32' is the pointee type.
+      // 'ptr<function,i32,read_write>' is the type of a pointer value that references
```

Comment:
are we forced to write `read_write` for all the pointers with this change?

Reply:
Only for storage buffers. Line 1806 effectively says if there is only one valid mode, then that is the default and it must not be specified. So this comment is for exposition only as it is invalid WGSL.

Reply:
With type inference for let-declarations, you never have to write this pointer type. (At least while pointer-to-storage is disallowed as in the baseline.  And afterward you only need to write it for formal parameter declaration.)